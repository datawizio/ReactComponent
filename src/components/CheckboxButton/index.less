@import "../../styles/variables";

button.button-checkbox {
  width: auto;
  padding: 0 5px;
  border-color: #d9d9d9;

  &.active {
    border-color: @primary-color !important;
    color: @primary-color !important;
  }
}