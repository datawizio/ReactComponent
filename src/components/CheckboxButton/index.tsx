import React, { useCallback, useContext, useRef } from "react";
import { Button } from "antd";
import clsx from "clsx";
import ConfigContext from "../ConfigProvider/context";
import { ButtonProps as AntButtonProps } from "antd/lib/button";

export interface CheckboxButtonProps extends Omit<AntButtonProps, "onChange"> {
  className?: string;
  checked?: boolean;
  text?: string;
  border?: boolean;
  highlight?: boolean;
  onChange?: (value: boolean) => void;
}

const CheckboxButton: React.FC<CheckboxButtonProps> = ({
  className,
  checked,
  text,
  border,
  highlight,
  onChange,
  ...props
}) => {
  const { translate } = useContext(ConfigContext);
  const btnRef = useRef<HTMLElement>(null);

  const buttonClasses = clsx(className, {
    "dw-btn": true,
    "button-checkbox": true,
    "active": checked,
    "no-border": !border,
    "highlight": highlight
  });

  const handleButtonClick = useCallback(() => {
    onChange && onChange(!checked);

    btnRef?.current?.blur();
  }, [checked, onChange]);

  return (
    <Button
      {...props}
      ref={btnRef}
      className={buttonClasses}
      onClick={handleButtonClick}
    >
      {text && translate(text)}
    </Button>
  );
};

CheckboxButton.defaultProps = {
  border: true
};

export default CheckboxButton;
