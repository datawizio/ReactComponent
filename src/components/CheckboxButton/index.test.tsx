import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import CheckboxButton, { CheckboxButtonProps } from "./index";

const mockProps: CheckboxButtonProps = {
  checked: false,
  text: "text",
  border: false,
  highlight: true,
  onChange: jest.fn()
};

const setUp = (props?) => mount(<CheckboxButton {...props} />);

describe("CheckboxButton component", () => {
  let component;
  beforeEach(() => {
    component = setUp();
  });

  it("CheckboxButton without props rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("CheckboxButton with props rendered correctly", () => {
    const wrapper = setUp(mockProps);
    expect(wrapper).toMatchSnapshot();
  });

  it("CheckboxButton with custom classname", () => {
    const wrapper = setUp({ ...mockProps, className: "custom-class-name" });
    expect(wrapper.find("Button").hasClass("custom-class-name")).toBeTruthy();
  });

  it("CheckboxButton change", () => {
    const wrapper = setUp(mockProps);
    const button = wrapper.find("Button");
    button.simulate("click");
    expect(wrapper.prop("onChange")).toBeCalledTimes(1);
  });
});
