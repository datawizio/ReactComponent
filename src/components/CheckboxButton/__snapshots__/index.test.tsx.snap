// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxButton component CheckboxButton with props rendered correctly 1`] = `
<CheckboxButton
  border={false}
  checked={false}
  highlight={true}
  onChange={[MockFunction]}
  text="text"
>
  <Button
    className="dw-btn button-checkbox no-border highlight"
    onClick={[Function]}
  >
    <Wave>
      <button
        className="ant-btn dw-btn button-checkbox no-border highlight"
        onClick={[Function]}
        type="button"
      >
        <LoadingIcon
          existIcon={false}
          loading={false}
          prefixCls="ant-btn"
        >
          <CSSMotion
            motionName="ant-btn-loading-icon-motion"
            onAppearActive={[Function]}
            onAppearStart={[Function]}
            onEnterActive={[Function]}
            onEnterStart={[Function]}
            onLeaveActive={[Function]}
            onLeaveStart={[Function]}
            removeOnLeave={true}
            visible={false}
          >
            <DomWrapper />
          </CSSMotion>
        </LoadingIcon>
        <span
          key=".0"
        >
          text
        </span>
      </button>
    </Wave>
  </Button>
</CheckboxButton>
`;

exports[`CheckboxButton component CheckboxButton without props rendered correctly 1`] = `
<CheckboxButton
  border={true}
>
  <Button
    className="dw-btn button-checkbox"
    onClick={[Function]}
  >
    <Wave>
      <button
        className="ant-btn dw-btn button-checkbox"
        onClick={[Function]}
        type="button"
      >
        <LoadingIcon
          existIcon={false}
          loading={false}
          prefixCls="ant-btn"
        >
          <CSSMotion
            motionName="ant-btn-loading-icon-motion"
            onAppearActive={[Function]}
            onAppearStart={[Function]}
            onEnterActive={[Function]}
            onEnterStart={[Function]}
            onLeaveActive={[Function]}
            onLeaveStart={[Function]}
            removeOnLeave={true}
            visible={false}
          >
            <DomWrapper />
          </CSSMotion>
        </LoadingIcon>
      </button>
    </Wave>
  </Button>
</CheckboxButton>
`;
